be1-go/channel/consensus/mod.go

```diff
@@ -341,53 +489,58 @@ func (c *Channel) processConsensusElectAccept(_ message.Message, msgData interfa
 
 	// check whether a message with the correct ID was received previously
 	_, valid := c.inbox.GetMessage(data.MessageID)
-
 	if !valid {
 		return xerrors.Errorf(messageNotReceived)
 	}
 
-	messageState, ok := c.messageStates[data.MessageID]
+	consensusInstance, ok := c.consensusInstances[data.InstanceID]
 	if !ok {
-		return xerrors.Errorf(messageStateInexistant, data.MessageID)
+		return xerrors.Errorf(consensusInexistant, data.InstanceID)
 	}
-	messageState.Lock()
-	defer messageState.Unlock()
+	consensusInstance.Lock()
+	defer consensusInstance.Unlock()
+
+	electInstance := consensusInstance.electInstances[data.MessageID]
+
+	if consensusInstance.decided || electInstance.failed {
+		return xerrors.Errorf(consensusFinished, data.InstanceID)
+	}
+
+	// Update the elect state
 	if data.Accept {
-		messageState.electAcceptNumber++
+		electInstance.positiveElectAccept++
+	} else {
+		electInstance.negativeElectAccept++
```

Comment:
There is no check if a previous ElectAccept was received with a different value or not.
This could allow a node to send both a positive and negative ElectAccept (even if it's not implemented or documented yet).
It woulnd't break anything but maybe it would make more sense to either disallow another ElectAccept from the same sender or either only count the last ?